src/elements/content-sharing/ContentSharing.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * @flow
+ * @file ContentSharing Element
+ * @author Box
+ */
+import React, { useCallback, useEffect, useState } from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import API from '../../api';
+import Internationalize from '../common/Internationalize';
+import ErrorMask from '../../components/error-mask/ErrorMask';
+import UnifiedShareModal from '../../features/unified-share-modal';
+import messages from '../../features/unified-share-modal/messages';
+import { convertItemResponse, convertUserResponse } from '../../features/unified-share-modal/utils/convertData';
+import {
+    CLIENT_NAME_CONTENT_SHARING,
+    FIELD_ALLOWED_INVITEE_ROLES,
+    FIELD_ENTERPRISE,
+    FIELD_HOSTNAME,
+    FIELD_ID,
+    FIELD_NAME,
+    FIELD_TYPE as FIELD_ITEM_TYPE,
+    FIELD_SHARED_LINK,
+    FIELD_SHARED_LINK_FEATURES,
+    TYPE_FILE,
+    TYPE_FOLDER,
+    FIELD_PERMISSIONS,
+    FIELD_EXTENSION,
+    FIELD_DESCRIPTION,
+} from '../../constants';
+import type { ItemType } from '../../common/types/core';
+import type { item as itemFlowType } from '../../features/unified-share-modal/flowTypes';
+import type { ContentSharingItemAPIResponse, ContentSharingSharedLinkType } from './types';
+
+type ContentSharingProps = {
+    apiHost: string,
+    itemID: string,
+    itemType: ItemType,
+    language: string,
+    showFormOnly?: boolean,
+    token: string,
+};
+
+const elementMessages = defineMessages({
+    loadingError: {
+        defaultMessage: 'Could not load shared link for this item.',
+        description: 'Message that appears when the ContentSharing Element cannot be loaded.',
+        id: 'be.contentSharing.loadingError',
+    },
+});
+
+function ContentSharing(props: ContentSharingProps) {
+    const { apiHost, itemID, itemType, language, token } = props;
+    const [item, setItem] = useState<?itemFlowType>(null);
+    const [sharedLink, setSharedLink] = useState<?ContentSharingSharedLinkType>(null);
+    const [currentUserID, setCurrentUserID] = useState<?string>(null);
+    const [errorExists, setErrorExists] = useState<boolean>(false);
```

Comment:
This might be a bit more robust if we store the error itself rather than a boolean for `errorExists`. This way in the future if we ever might display different error messages this won't need to be redone. Is that a possibility?